### PR TITLE
PIP-1167-apply-blacklists

### PR DESCRIPTION
**Pull request by ottojolanki**

### CODE REVIEW

File: templates/json/GRCh38/36/references_template.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
         "build_mappable_only_bed": "large",
         "get_chrom_info": "large",
         "get_chrom_sizes": "large",
-        "get_center_sites": "large"
+        "get_center_sites": "large",
+        "subtract_blacklists_from_mappable_regions": "large:"
```
Comment (keenangraham) [inline templates/json/GRCh38/36/references_template.json:12]: `"large:"`

File: wdl/tasks/bedops.wdl [hunk 2]
```diff
@@ -167,3 +167,32 @@ task merge {
         disks: resources.disks
     }
 }
+
+
+task difference {
+    input {
+        Array[File] subtrahends
+        File minuend
```
Comment (keenangraham) [inline wdl/tasks/bedops.wdl:175]: Lol, had to look these up.

File: references.wdl [hunk 3]
```diff
@@ -78,6 +80,17 @@ workflow references {
         build_mappable_only_bed.mappable_regions
     ])
 
+    if (defined(blacklists)) {
+        call subtract.subtract_blacklists_from_mappable_regions {
+            input:
+                blacklists=select_first([blacklists]),
```
Comment (keenangraham) [inline references.wdl:86]: For consistency:
```
blacklists=select_first([
    blacklists
]),
```

File: references.wdl [hunk 4]
```diff
@@ -78,6 +80,17 @@ workflow references {
...
+                mappable_regions=mappable_regions_output,
+                resources=compute.runtimes[machine_sizes.subtract_blacklists_from_mappable_regions],
+        }
+    }
+
+    File mappable_regions_processed = select_first([subtract_blacklists_from_mappable_regions.mappable_regions_subtracted, mappable_regions_output])
```
Comment (keenangraham) [inline references.wdl:92]:
```
select_first([
    subtract_blacklists_from_mappable_regions.mappable_regions_subtracted,
    mappable_regions_output
])
```

**Comment by keenangraham** [comment]
Test failure. `test_references`

File: templates/json/GRCh38/36/references_template.json (hunk 1 above)
Comment (ottojolanki) [reply templates/json/GRCh38/36/references_template.json:12]: Thanks

File: wdl/tasks/bedops.wdl (hunk 2 above)
Comment (ottojolanki) [reply wdl/tasks/bedops.wdl:175]: Yeah, was pretty stoked to actually have a reason to use these :)

**Pull request merged**